endpoints.php

```diff
@@ -1491,4 +1507,65 @@ function getQuestionHasAnswers($request)
         return new WP_Error($e->getCode(), $e->getMessage(), $e->getStatus());
     }
     
+}
+
+function upload_videos_to_kaltura()
+{
+    if (LOAD_KALTURA) {
```

Comment:
Before allowing this to be run, we should check and make sure this operation is not running already. You can save an option named `tapestry_kaltura_upload_in_progress` for instance using the Wordpress [update_option()](https://developer.wordpress.org/reference/functions/update_option/) function (set it to true at the beginning and to false after done) and retrieve it using get_option() (making sure it's not running, otherwise stopping automatically). We should make sure to handle timeouts as well so it's not "stuck" in upload mode forever.

Reply:
I've added some code to do this, although it currently does not handle timeouts as far as I know.